datacube_stats/main.py

```diff
@@ -819,40 +836,56 @@ def __call__(self, index, sources_spec, date_ranges):
         for time_period in date_ranges:
             _LOG.info('Making output product tasks for time period: %s', time_period)
             timer = MultiTimer().start('creating_tasks')
+            created_tasks = 0
 
-            # Tasks are grouped by tile_index, and may contain sources from multiple places
-            # Each source may be masked by multiple masks
-            tasks = {}
-            for source_spec in sources_spec:
-                group_by_name = source_spec.get('group_by', DEFAULT_GROUP_BY)
-                source_filter = source_spec.get('source_filter', None)
+            if self.cell_index is not None:
```

Comment:
`self.cell_index` is not a good name anymore, has to be updated to indicate that it is now a collection of indices not  just a single one. If I search a file for all occurrences of a specific name I should expect all use-sites to be of the same type/contain similar things. As it stands now, name `cell_index` sometimes refers to actual "cell index" and some times to "a list of cell indices or None".

Might need to update names up along the call stack as well.